heron/tools/cli/src/python/execute.py

```diff
@@ -63,14 +63,17 @@ def heron_class(class_name, lib_jars, extra_jars=None, args=None, java_defines=N
   heron_env['HERON_OPTIONS'] = opts.get_heron_config()
 
   # print the verbose message
-  Log.debug('$> %s' % ' '.join(all_args))
-  Log.debug('Heron options: %s' % str(heron_env["HERON_OPTIONS"]))
+  Log.debug("Invoking class using command:\n``%s''", ' '.join(all_args))
+  Log.debug('Heron options: {%s}', str(heron_env["HERON_OPTIONS"]))
 
   # invoke the command with subprocess and print error message, if any
-  status = subprocess.call(all_args, env=heron_env)
-  if status != 0:
-    err_str = "User main failed with status %d. Bailing out..." % status
-    raise RuntimeError(err_str)
+  proc = subprocess.Popen(all_args, env=heron_env, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+  stdout, stderr = proc.communicate()
+  retcode = proc.returncode
+  Log.debug('Dumping stderr from process:\n%s', stderr[:-1])
+  if retcode != 0 and stdout:
+    Log.error(stdout[:-1])
+  return proc.returncode
```

Comment:
instead of logging error messages and returning the return code, could we return some sort of response object that includes he return code and the message?